tests/test_schema.py

```diff
@@ -2099,6 +2099,44 @@ class Model(BaseModel):
     }
 
 
+def test_multiple_models_with_same_name(create_module):
+    module = create_module(
+        # language=Python
+        """
+from pydantic import BaseModel
+
+
+class ModelOne(BaseModel):
+    class NestedModel(BaseModel):
+        a: float
+
+    nested: NestedModel
+
+
+class ModelTwo(BaseModel):
+    class NestedModel(BaseModel):
+        b: float
+
+    nested: NestedModel
+
+
+class NestedModel(BaseModel):
+    c: float
+        """
+    )
+
+    models = [module.ModelOne, module.ModelTwo, module.NestedModel]
+    model_names = set(schema(models)['definitions'].keys())
+    expected_model_names = {
```

Comment:
Could this test `get_model_name_map` directly? It'd be the same structure as `test_model_name_maps`.

Reply:
I tend to think that what really matters in the end is just to have the right definitions in the schema. We could refactor `get_model_name_map` and still have the same result. It's always the question of "unit test" and if you simply test your function or the expected output. Here IMO testing the output is what has the most value

Reply:
You're right, testing the API is always best. I just thought that maybe `test_model_name_maps` was part of the API, but I too encountered the problem via `schema`, so you're right that testing that may make more sense.